lib/src/main/kotlin/com/swmansion/starknet/service/http/OkhttpHttpService.kt

```diff
@@ -0,0 +1,96 @@
+package com.swmansion.starknet.service.http
+
+import com.swmansion.starknet.service.http.HttpService.Payload
+import com.swmansion.starknet.service.http.handlers.HttpErrorHandler
+import okhttp3.Call
+import okhttp3.Callback
+import okhttp3.HttpUrl.Companion.toHttpUrl
+import okhttp3.MediaType.Companion.toMediaType
+import okhttp3.OkHttpClient
+import okhttp3.RequestBody.Companion.toRequestBody
+import okhttp3.Response
+import java.io.IOException
+import java.util.concurrent.CompletableFuture
+
+/**
+ * Service for making http requests.
+ */
+internal class OkhttpHttpService(private val errorHandler: HttpErrorHandler) : HttpService {
```

Comment:
Why do we need to inject error handling here?

Reply:
Can't we inject (optionally) OkHttpClient instead to check handling of invalid requests?